### DISCUSSION TITLE
How to handle new issues and branches in this project

### DISCUSSION THREAD

**Discussion by ksvedal** [ksvedal/uu-chrome-extension#33] [Announcements]
# Issues

Don't assign to anyone when first setting up issues. Unless you're going to work on it immediately.
Remember `Labels`, `Projects`, `Milestone` when setting up (this will automatically add them to the issue board).

# Branches

Main branch has protection against direct pushing. It has to happen via merge from head branches created through issue assigning which requires one reviewers approval.

On assign will create a new branch unless the issue has the "antibranch" label

Then:

1. `git fetch` or `git branch` and you might see the new branch.
2. `git checkout [branch name]`
3. Make your changes.
4. `git add`
5. `git commit`
6. `git pull origin main` (this will keep merge conflicts small and only local with the changes you’ve made yourself)
7. Fix merge conflicts
8. `git status` to check that its good.
9. `git push`

Go back to github.no and create pull request then get reviewer to merge.

Branch will merge and self-delete.

**Comment by dristanta-silwal** [comment]
Thanks for addressing this question.
